### PR TITLE
Updated section "backward compatibility" in contributing.rst

### PR DESCRIPTION
**Pull request by axel-lauer**
## Description

This PR adds a reference to the new "backward compatibility policy" to the documentation (contributing.rst, Backward compatibility: https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility).

This PR should be merged together with PR https://github.com/ESMValGroup/ESMValTool/pull/2879, which introduces new backward compatibility policy to ESMValTool documentation.

Closes discussion: https://github.com/ESMValGroup/Community/discussions/7

Link to [documentation](https://esmvaltool--1918.org.readthedocs.build/projects/ESMValCore/en/1918/)

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

### CODE REVIEW

**Comment by valeriupredoi** [comment]
@alistairsellar we're starting to get our ducks in line for the impending 2.8 release, mate - this one is marked as M2.8 but it's still a draft, you reckon you can make it RfR soon please? Cheers :beer:

**Comment by alistairsellar** [comment]
OK @valeriupredoi is now ready after one final change.

As per @axel-lauer's  description, this PR links with Tool doc changes (ESMValGroup/ESMValTool#2879), and so doesn't necessarily make sense on its own (it possibly introduces broken links without the linked PR).

Also, not sure if the new references should be of the form `<esmvaltool:backward_compatibility>` rather than full URLs? I'm not confident enough in that to change Axel's commits.

**Comment by valeriupredoi** [comment]
many thanks @alistairsellar :beer: A few pointy points:

- I tried adding @axel-lauer as reviewer here but GH has troubles finding him in the list of contributors at the Rview suggestions tab - something's fishy there
- @alistairsellar would you mind checking the doc build failed test and the PR description boxes please?
- please tag @remi-kazeroni as reviewer too since he'll be our Release Manager this release cycle :beer:

**Comment by remi-kazeroni** [comment]
Hi all, the start of the release process is approaching (next week). It would be great to have this in. Since @axel-lauer started the PR, GitHub does let the author to be the reviewer. Axel, feel free to comment here if the PR looks good to you. @alistairsellar, I'd be happy to take a final look at this if you wish. Feel free to assign me and anyone who could/should have a look at this PR. Thanks 👍

**Comment by alistairsellar** [comment]
Thanks @valeriupredoi. I've now looked into the docs build error and can't figure the cause. Please could you, @axel-lauer  @remi-kazeroni advise? 

The error message is very generic and doesn't tip me off about the cause: 
```
ERROR 1: PROJ: proj_create_from_database: Open of /home/docs/checkouts/readthedocs.org/user_builds/esmvalcore/conda/1918/share/proj failed
```

Also two warnings in the output about cross-references - could these be the underlying cause of the failure?:
```
/home/docs/checkouts/readthedocs.org/user_builds/esmvalcore/checkouts/1918/doc/contributing.rst:142: WARNING: undefined label: esmvaltool:backward_compatibility
/home/docs/checkouts/readthedocs.org/user_builds/esmvalcore/checkouts/1918/doc/contributing.rst:586: WARNING: undefined label: esmvaltool:backward_compatibility
```
If so, they may arise from the fact that the doc does not yet exist in the esmvaltool documentation? They will resolve correctly when ESMValGroup/ESMValTool#2879 is merged.

**Comment by valeriupredoi** [comment]
cheers @alistairsellar - it's those two warnings, the build is set to fail on warnings - would you be ok to fix those pls :beer:

**Comment by alistairsellar** [comment]
Hi @valeriupredoi, I *think* (though not certain and welcome an alternative hint) that the warnings are because the referenced document does not *yet* exist in the esmvaltool documentation. If I'm right, then the only way to fix them is to merge ESMValGroup/ESMValTool#2879.

**Comment by valeriupredoi** [comment]
@alistairsellar it looks that way indeed - is Tool/2879 ready to be merged? I just dumped `main` into it so that should sort out the tests - if you OK with that I can approve and merge when tests finish green :green_circle:

**Comment by alistairsellar** [comment]
@valeriupredoi, https://github.com/ESMValGroup/ESMValTool/pull/2879 has now been merged, and build now succeeding (after a final fix to links). So I think this PR is ready to merge.

**Comment by alistairsellar** [comment]
... pending review. I've tagged @remi-kazeroni following his kind offer above.

**Comment by alistairsellar** [comment]
Thanks @valeriupredoi. Merged in main.

Sorry, I need you to explain "boxes need a sweep" in words of one syllable. Should I remove the checks that are not relevant, or tick the boxes that I believe are done (not sure whether developer or reviewer is supposed to do that), or both, or perhaps something else?

**Comment by valeriupredoi** [comment]
hahah let me do that for you now, you have a look, and next time you hear "clean up boxes" (in this context, not when your wife asks you to do that at home), you'll know what that means :grin:

**Review by valeriupredoi — APPROVED** [review]
cheers muchly @alistairsellar :beer: @remi-kazeroni @bouweandela you guys have a wee bit of time have a last look at this and merge, plese?

EDIT: my apologies @alistairsellar - just noticed this was @axel-lauer 's PR not yours, sorry to have put you through the paces to clean up PR descriptions - it's the PR OP that needs to do that :grin:

File: doc/contributing.rst
Comment (bouweandela) [inline doc/contributing.rst:801]: It would be good to update this section so it contains this information: https://docs.esmvaltool.org/en/latest/community/release_strategy.html#add-release-notes

File: doc/contributing.rst
Comment (remi-kazeroni) [inline doc/contributing.rst:638]:
```suggestion
You should tag the `@ESMValGroup/esmvaltool-coreteam`_ to
```

File: doc/contributing.rst
Comment (remi-kazeroni) [inline doc/contributing.rst:154]: The link is currently not clickable
```suggestion
  #263 <https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/263>`__).
```

**Review by remi-kazeroni — CHANGES_REQUESTED** [review]
Thanks for taking care of this topic @alistairsellar and @axel-lauer! This looks good to me. I just have 2 very minor suggestions for improvement. 

Good to have our backward compatibility policy fully documented. Let's make good use of that to draft the Changelog for the upcoming release 👍

File: doc/contributing.rst
Comment (alistairsellar) [reply doc/contributing.rst:801]: Good point. Now done in [bdd914f](https://github.com/ESMValGroup/ESMValCore/pull/1918/commits/bdd914fcdaa54214f94d12760a96c737c77e3d56)

**Comment by alistairsellar** [comment]
Righteeho, I've resolved Bouwe's request, and I think that Remi's requests have been folded in. So I *think* all we need is for @remi-kazeroni to mark as approved and then we're good to merge...

**Comment by valeriupredoi** [comment]
cheers, bud @alistairsellar :beer:

**Review by remi-kazeroni — APPROVED** [review]
Thanks everyone!

**Pull request merged**
